provision.py

```diff
@@ -173,6 +175,17 @@ def main():
     sh.do_destroy_rebuild_database(**LOUD)
     sh.postgres_init_test_db(**LOUD)
     sh.do_destroy_rebuild_test_database(**LOUD)
+
+    #Checking if the version of npm is sufficient
+    neededVersion = '2.14.20'
+    versionOutput = subprocess.Popen(["npm", "--version"], stdout = subprocess.PIPE)
+    version = versionOutput.communicate()[0].strip('\n')
+    compareVersion = StrictVersion(neededVersion) > StrictVersion(version)
+    if compareVersion == True:
+        print("The version of npm on your system is not sufficient")
+        print("Installing the latest version.....")
+        os.system("sudo npm -g install npm")
+
```

Comment:
A few comments:
* the code looks correct
* The variable names you're using in camelCase are java-style; you should use python style variable names (delimited by underscores, e.g. python_case) in your Python code.
* You should use `subprocess.check_output` rather than Popen + communicate to simplify the code.

Otherwise, this looks reasonable!  There will be some merge conflicts with a branch I'm working on; I can take care of resolving those whichever one lands first.